server/lib/providers/bitbucket.js

```diff
@@ -16,110 +16,84 @@ const bitbucket = () =>
     rest_version: '2.0'
   });
 
-const checkRepo = (repository) =>
-  new Promise((resolve, reject) => {
-    try {
-      const { user, repo } = utils.parseRepo(repository);
-
-      bitbucket().get('repositories/{username}/{repo_slug}', { username: user, repo_slug: repo }, (err) => {
-        if (err) {
-          return reject(err);
-        }
-
-        return resolve({ user, repo });
-      });
-    } catch (e) {
-      reject(e);
-    }
-  });
+const checkRepo = (repository) => {
+  try {
+    const { user, repo } = utils.parseRepo(repository);
+
+    return bitbucket()
+      .get('repositories/{username}/{repo_slug}', { username: user, repo_slug: repo })
+      .then(() => ({ user, repo }));
+  } catch (e) {
+    return Promise.reject(e);
+  }
```

Comment:
Since you're working with `Promises` inside the `try`/`catch`, you should just omit the `try`/`catch`. If an error occurs within the promise chain starting at line 23, it won't actually be thrown anyway; it'll be returned as part of the `Promise` resolution.

Reply:
`utils.parseRepo(repository)` can throw an error.